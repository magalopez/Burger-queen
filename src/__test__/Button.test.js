import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import Button from "../components/generic_components/button";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Button text="Testing" />);
  const firstRender = asFragment();
  expect(firstRender).toMatchSnapshot();
});

const testClick = jest.fn(() => {
  return <p>You Clicked</p>;
});

it("insert text in button", () => {
  const { getByTestId, getByText } = render(
    <Button text="Testing" onclick={testClick} className={"none"} />
  );
  expect(getByTestId("buttontag")).toHaveTextContent(/Testing/i);
  expect(getByText("Testing")).not.toHaveClass("none");

  fireEvent.click(getByTestId("buttontag"));
});
