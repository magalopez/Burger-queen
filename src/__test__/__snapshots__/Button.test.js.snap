// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="field is-grouped"
  >
    <button
      data-testid="buttontag"
    >
      Testing
    </button>
  </div>
</DocumentFragment>
`;
